text/0000-deprecate-string-based-actions.md

```diff
@@ -0,0 +1,89 @@
+- Start Date: 2020-05-23
+- Relevant Team(s): Ember.js
+- RFC PR: (after opening the RFC PR, update this with a link to it and update the file name)
+- Tracking: (leave this empty)
+
+# Deprecate String Based Actions
+
+## Motivation
+
+> In Ember 1.13 we introduced [closure actions](https://github.com/emberjs/rfcs/blob/00ac2685c86f27d41547012903f485a4ef338d27/active/0000-improved-actions.md) have been recommended over string based actions. This officially deprecates string based actions in favor of closure actions,
```

Comment:
FWIW, I think the general phrasing here is a bit off. I don't think we are deprecating string based actions in favor of closure actions so much as we are deprecating them in favor of passing functions. Yes, this is what `(action` returns, but we _really_ don't want folks using `(action` at all. They should use the `@action` decorator along with `fn` (when you need to curry an argument).

Reply:
We are not currently deprecating use of the `action` helper with a function.

Reply:
Ya, exactly, that is why that isn't what I said 😛. 

I'm saying that we are deprecating using any form of `action` (either helper or modifier form) when passing a string, but the overall rationale is that it is much easier to "just pass a function". In the Octane programming model, you don't use `{{action` or `(action`.

Reply:
While it is possible to use other modifiers with classic Ember components, the need to use a kind of syntax not shown in any guides makes this less ergonomic than just continuing to use the `action` modifier IMO. It will also be significantly more difficult to codemod.

Reply:
I’m not really following the logic here. What is the syntax that you are referring to (that isn’t in the rest of the guides)? Both decorators (if that is what you meant) and modifiers (if that is what you meant) are used heavily in the guides...

Reply:
The `action(function() {...})` syntax

Reply:
Sure, but by that logic we should avoid `.extend()` here too since the guides don’t cover it either. 😜🤷‍♂️